astropy/stats/histogram.py

```diff
@@ -207,11 +206,10 @@ def freedman_bin_width(data, return_bins=False):
 
     if return_bins:
         dmin, dmax = data.min(), data.max()
-        if dx < 1e-6:
-            dx = median_absolute_deviation(data)
-        if dx < 1e-6:
-            raise ValueError("data has too small IQR and median_absolute_deviation values")
-        Nbins = max(1, np.ceil((dmax - dmin) / dx))
+        d_minmax = dmax - dmin
+        if dx <= (1e-6 * d_minmax):
+            raise ValueError("data has too small IQR")
```

Comment:
@jakevdp , being someone familiar with this algorithm, do you have any recommendation on an ideal error message? How about this?

"Interquartile range (IQR) is too small. Please ensure Delta_b > 1e-6 (MAX - MIN)."

Reply:
I think we should put ourselves in the shoes of a user who doesn't know what this algorithm does, but was told it's a good way to create a histogram. In that case, "Please ensure Delta_b > 1e-6 (MAX - MIN)" is not particularly helpful. I'd probably go with a message something like this:

```
raise ValueError("The inter-quartile range of the data is too small: it indicates a histogram "
                 "with more than 1,000,000 bins. Please use another bin method, such as "
                 "bins='scott'")
```

I wonder if we might also make the cutoff a user-specified parameter. For example, it would be quite annoying if a user actually wanted to create a histogram with >1,000,000 bins, and the code offered no way to do that because of a hard-coded "magic number".

Reply:
The other option (probably better, IMHO) would be to not have any hard-coded parameter, but to raise an error if the IQR is actually **equal** to zero. That would address the case of the bug that prompted this fix, without making any additional assumptions about what the user intends to do with the output of the function.

Reply:
Well... If you want to avoid magic number, we can simply `try ... except ValueError ...` to catch it after the fact, and re-raise with your error (but without specific value). The original error was from `np.arange` and it appears to be memory-limited.

Reply:
As the original error was memory related, a try and except would probably be the best option as that removes the need for a user parameter and catches something that might have slightly different behaviors on different machines.

Reply:
Just so we're clear, re-raising in the `except` block will have longer traceback with, "During handling of the above exception, another exception occurred..." Some users have found this confusing in the past. But I'll add a commit to implement this as requested.